common/src/main/java/com/microsoft/identity/common/internal/net/UrlConnectionHttpClient.java

```diff
@@ -319,10 +323,17 @@ private static void safeCloseStream(@Nullable final Closeable stream) {
     }
 
     private HttpResponse executeHttpSend(HttpRequest request, Consumer<HttpResponse> completionCallback) throws IOException {
+        final String requestMethod = request.getRequestMethod();
         final HttpURLConnection urlConnection = setupConnection(request);
-        urlConnection.setRequestMethod(request.getRequestMethod());
+        urlConnection.setRequestMethod(requestMethod);
         urlConnection.setUseCaches(true);
-        setRequestBody(urlConnection, request.getRequestContent(), request.getmRequestHeaders().get(HttpConstants.HeaderField.CONTENT_TYPE));
+
+        if (!requestMethod.equalsIgnoreCase(GET.name())
+                && !requestMethod.equalsIgnoreCase(TRACE.name())
+                && !requestMethod.equalsIgnoreCase(OPTIONS.name())
+                && !requestMethod.equalsIgnoreCase(HEAD.name())) {
```

Comment:
At this point, we have an enum, and we should just expose it and create a method on it that describes what's happening rather than perform a bunch of string compare.

Reply:
You know what, we can do that later.